instrumentation/opentelemetry-instrumentation-flask/tests/test_flask_compatibility.py

```diff
@@ -0,0 +1,373 @@
+            def stream_cleanup_error_endpoint():
+                def generate():
+                    yield "Data"
+
+                return flask.Response(flask.stream_with_context(generate()))
+
+            client = app.test_client()
+
+            # The request should still complete successfully despite cleanup errors
+            response = client.get("/stream_cleanup_error")
+            self.assertEqual(response.status_code, 200)
+            self.assertEqual(response.data, b"Data")
+
+    def test_flask_31_streaming_context_cleanup(self):
+        """Test that Flask 3.1+ streaming responses have proper context cleanup to prevent token reuse"""
+        # Skip this test on Python < 3.10 due to context management compatibility issues
+        if sys.version_info < (3, 10):
+            self.skipTest(
+                "Flask 3.1+ streaming context cleanup only enabled on Python 3.10+"
+            )
+        # Also skip if Flask version is < 3.1
+        if package_version.parse(flask.__version__) < package_version.parse(
+            "3.1.0"
+        ):
+            self.skipTest(
+                "Flask 3.1+ streaming context cleanup requires Flask 3.1+"
+            )
+        app = flask.Flask(__name__)
+        FlaskInstrumentor().instrument_app(app)
+
+        @app.route("/stream_flask31")
+        def flask31_streaming_endpoint():
+            def generate():
+                for chunk_idx in range(10):
+                    yield f"Chunk {chunk_idx}\n"
+
+            return flask.Response(flask.stream_with_context(generate()))
+
+        @app.route("/stream_normal")
+        def stream_normal_endpoint():
+            def generate():
+                for chunk_idx in range(5):
+                    yield f"Normal {chunk_idx}\n"
+
+            return flask.Response(flask.stream_with_context(generate()))
+
+        client = app.test_client()
+
+        # Make multiple consecutive streaming requests to test for token reuse issues
+        for request_idx in range(20):
+            response = client.get("/stream_flask31")
+            self.assertEqual(response.status_code, 200)
+            expected_data = "".join(f"Chunk {j}\n" for j in range(10)).encode()
+            self.assertEqual(response.data, expected_data)
+
+        # Test normal streaming requests as well
+        for request_idx in range(5):
+            response = client.get("/stream_normal")
+            self.assertEqual(response.status_code, 200)
+            expected_data = "".join(f"Normal {j}\n" for j in range(5)).encode()
+            self.assertEqual(response.data, expected_data)
+
+        # Mix of streaming requests
+        for request_idx in range(10):
+            endpoint = (
+                "/stream_flask31" if request_idx % 2 == 0 else "/stream_normal"
+            )
+            response = client.get(endpoint)
+            self.assertEqual(response.status_code, 200)
+
+        if package_version.parse(flask.__version__) >= package_version.parse(
```

Comment:
This check is already handled by the skip before right? Anyway in order to test this we need to add a `test-requirements-3.txt` with flask 3.1 and adjust `tox.ini`. That would require to run `tox -e generate-workflows` and apply the changes for the CI to run green

Reply:
> @fatelei you still need to add a new test requirements file for flask 3.1 as the comment above

ok